### PR TITLE
generate-docs: added pack seeker

### PR DESCRIPTION
**Pull request by jochman**
## Status
- [x] In Progress
- [ ] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: https://github.com/demisto/etc/issues/22759

## Description
if the file is a part of pack. will unify the integration/script first.

## Screenshots
image

## Must have
- [x] Tests
- [ ] Documentation

### CODE REVIEW

File: demisto_sdk/__main__.py [hunk 1]
```diff
@@ -541,6 +541,9 @@ def init(**kwargs):
     "-o", "--output", help="The output dir to write the documentation file into,"
                            " documentation file name is README.md. If not specified, will be in the yml dir.",
     required=False)
+@click.option(
+    "--pack", help="A non-unified pack (code and yml in separate files). Works only with scripts and integrations.",
```
Comment (Itay4) [inline demisto_sdk/__main__.py:545]: lets remove that flag, keep only `-i` and detect if its a pack

File: demisto_sdk/tests/integration_tests/generate_docs_integration_test.py [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+    - Running validation on it.
+
+    Then
+    - Ensure validation passes.
+    - Ensure success validation message is printed.
+    """
+    test_file = 'demisto_sdk/tests/test_files/content_repo_example/Packs/FeedAzure/Integrations/FeedAzure/FeedAzure.yml'
+    test_file_path = join(git_path(), test_file)
+    runner = CliRunner(mix_stderr=False)
+    result = runner.invoke(main, [GENERATE_DOCS_COMMAND, '-i', test_file_path])
```
[the remark this answers is not in the document]
Comment (jochman) [reply demisto_sdk/tests/integration_tests/generate_docs_integration_test.py:25]: it's a yml in a pack, so if it's a part of a pack it's automatically unifies it.

[the remark this answers is not in the document]
Comment (jochman) [reply demisto_sdk/tests/integration_tests/generate_docs_integration_test.py:25]: if you want to just give a directory, we should open a new PR with enhancement to Unifier

File: demisto_sdk/__main__.py [hunk 3]
```diff
@@ -603,6 +604,16 @@ def generate_doc(**kwargs):
         return 1
 
     print(f'Start generating {file_type} documentation...')
+    full_dir_name = os.path.dirname(os.path.realpath(input_path))
+    if is_file_path_in_pack(full_dir_name, check_substring=True):
+        unifier = Unifier(full_dir_name, output=full_dir_name)
```
[the remark this answers is not in the document]
Comment (jochman) [reply demisto_sdk/__main__.py:609]: you're giving a specific yml to it. 
If there're multiple ymls, it will not work as we need to find which yml file is the right one.

File: demisto_sdk/__main__.py [hunk 4]
```diff
@@ -603,6 +604,16 @@ def generate_doc(**kwargs):
         return 1
 
     print(f'Start generating {file_type} documentation...')
```
[the remark this answers is not in the document]
Comment (jochman) [reply demisto_sdk/__main__.py:606]: because it works when suppling a yml.
I can search for a yml in the directory.

File: CHANGELOG.md [hunk 5]
```diff
@@ -1,5 +1,6 @@
 # Changelog
 * Added *-a , --validate-all* option in *validate* to run all validation on all files.
+* Added *-i, --input* option in *generate-docs* to run on a specific file.
```
Comment (Itay4) [inline CHANGELOG.md:3]: thats not accurate, `-i` was already there, you added support for packs

File: demisto_sdk/commands/generate_docs/README.md [hunk 6]
```diff
@@ -5,7 +5,7 @@ Generate a README file for an Integration, Script or a Playbook.
 This command is used to create a documentation file for Demisto content files.
 
 **Arguments**:
-* **-i, --input**Path of the yml file.
+* **-i, --input**Path of the yml file. If in pack, will unify the file into a unified file.
```
Comment (Itay4) [inline demisto_sdk/commands/generate_docs/README.md:8]: update that it supports packs as input

File: demisto_sdk/__main__.py [hunk 7]
```diff
@@ -577,6 +578,17 @@ def generate_doc(**kwargs):
     insecure = kwargs.get('insecure')
     verbose = kwargs.get('verbose')
 
+    if input_path and os.path.isdir(input_path):
+        print("Input is a directory. will try to find yml files.")
```
Comment (Itay4) [inline demisto_sdk/__main__.py:582]:
```suggestion
        print("Input is a directory. Looking for YAML files...")
```

File: demisto_sdk/commands/common/tools.py [hunk 8]
```diff
@@ -731,3 +745,61 @@ def is_file_from_content_repo(file_path: str) -> Tuple[bool, str]:
         return True, '/'.join(input_path_parts[len(content_path_parts):])
     else:
         return False, ''
+
+
+class FileFinder:
+    class Extension:
```
Comment (Itay4) [inline demisto_sdk/commands/common/tools.py:751]: lets set these to be consts in the consts file and remove them from here

File: demisto_sdk/commands/common/tools.py [hunk 9]
```diff
@@ -731,3 +745,61 @@ def is_file_from_content_repo(file_path: str) -> Tuple[bool, str]:
         return True, '/'.join(input_path_parts[len(content_path_parts):])
     else:
         return False, ''
+
+
+class FileFinder:
```
Comment (Itay4) [inline demisto_sdk/commands/common/tools.py:750]: this class kind of breaks the standard of this file, and its not like actually its needed, lets stick to single functions and drop it

File: demisto_sdk/__main__.py [hunk 10]
```diff
@@ -577,6 +578,17 @@ def generate_doc(**kwargs):
...
+        file_list = FileFinder.get_yml_files_in_dir(input_path)
+        if not file_list:
+            print_error(f"Could not find any .yml files in {input_path}")
```
Comment (Itay4) [inline demisto_sdk/__main__.py:585]:
```suggestion
            print_error(f"Could not find any YAML files in {input_path}")
```

File: demisto_sdk/__main__.py [hunk 11]
```diff
@@ -577,6 +578,17 @@ def generate_doc(**kwargs):
...
+            return 1
+        if len(file_list) > 1:
+            print_error(f"Found more than 1 .yml file in '{input_path}' which is not supported.")
```
Comment (Itay4) [inline demisto_sdk/__main__.py:588]: why not.. we're losing the whole point here..

File: demisto_sdk/__main__.py [hunk 12]
```diff
@@ -577,6 +578,17 @@ def generate_doc(**kwargs):
...
+            return 1
+        input_path = file_list[0]
+        print(f"Found yml file! {input_path}")
```
Comment (Itay4) [inline demisto_sdk/__main__.py:591]:
```suggestion
        print(f"Found YAML file: {input_path}\n Generating documentation for it...")
```

File: demisto_sdk/__main__.py [hunk 13]
```diff
@@ -596,13 +608,22 @@ def generate_doc(**kwargs):
...
             return 1
-
-    file_type = find_type(kwargs.get('input', ''))
+    file_type = find_type(input_path)
     if file_type not in ["integration", "script", "playbook"]:
         print_error('File is not an Integration, Script or a Playbook.')
         return 1
 
     print(f'Start generating {file_type} documentation...')
+    full_dir_name = os.path.dirname(os.path.realpath(input_path))
```
Comment (Itay4) [inline demisto_sdk/__main__.py:617]: why is this needed? you already get this as input

File: demisto_sdk/__main__.py [hunk 14]
```diff
@@ -596,13 +608,22 @@ def generate_doc(**kwargs):
...
+    if is_file_path_in_pack(full_dir_name, check_substring=True):
```
Comment (Itay4) [inline demisto_sdk/__main__.py:618]: what exactly are we checking here? that we got a pack ?

File: demisto_sdk/commands/unify/unifier.py [hunk 15]
```diff
@@ -134,13 +141,7 @@ def merge_script_package_to_yml(self):
         """Merge the various components to create an output yml file
         """
         print("Merging package: {}".format(self.package_path))
-        package_dir_name = os.path.basename(self.package_path)
```
Comment (Itay4) [inline demisto_sdk/commands/unify/unifier.py:137]: why did we need to move this? why cant we just call this function from generate-docs ?
